runtime/include/qio/qbuffer.h

```diff
@@ -500,13 +485,28 @@ typedef chpl_bool qio_bool;
 #define qio_realloc(ptr, size) sys_realloc(ptr, size)
 #define qio_memalign(boundary, size) sys_memalign(boundary, size)
 #define qio_free(ptr) sys_free(ptr)
-#define qio_strdup(ptr) strdup(ptr)
 #define qio_memcpy(dest, src, num) memcpy(dest, src, num)
 
 typedef bool qio_bool;
 
 #endif
 
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static inline char* qio_strdup(const char* ptr)
+{
+  char* ret = (char*) qio_malloc(strlen(ptr)+1);
+  if( ret ) strcpy(ret, ptr);
+  return ret;
+}
+
+#ifdef __cplusplus
+} // end extern "C"
+#endif
+
```

Comment:
This is fine with me but looks really familiar... I thought I made this change somewhere. Anyway it can be in this PR.